aws_lambda_powertools/event_handler/util.py

```diff
@@ -18,6 +18,27 @@ def __hash__(self):
         return hash(frozenset(self.keys()))
 
 
+class _FrozenListDict(List[Dict[str, List[str]]]):
+    """
+    Freezes a list of dictionaries containing lists of strings.
+
+    This function takes a list of dictionaries where the values are lists of strings and converts it into
+    a frozen set of frozen sets of frozen dictionaries. This is done by iterating over the input list,
+    converting each dictionary's values (lists of strings) into frozen sets of strings, and then
+    converting the resulting dictionary into a frozen dictionary. Finally, all these frozen dictionaries
+    are collected into a frozen set of frozen sets.
+
+    This operation is useful when you want to ensure the immutability of the data structure and make it
+    hashable, which is required for certain operations like using it as a key in a dictionary or as an
+    element in a set.
+
+    Example: [{"TestAuth": ["test", "test1"]}]
+    """
+
+    def __hash__(self):
+        return hash(frozenset({_FrozenDict({key: frozenset(self) for key, self in item.items()}) for item in self}))
```

Comment:
break this into multiple lines to make it easier to read plz

Reply:
I understand what you mean, but black is formatting this line, so it makes sense to follow the linter formatting. WDYT?

Reply:
use multiple expressions in this case. I'm gonna pull the PR to send a suggestion as Web version is tricky

Reply:
Best I could tinker right now - got rid of the extra wrapping as it was unnecessary. Tests passing but lemme know otherwise.

I broke by:

1. Group all items that are hashable
2. We're interested in immutable values so only use `frozenset` in the value itself
3. Create a `frozenset` out of all hashable items
4. Hash it

That way we also don't need to use `_FrozenDict` to wrap some of it too.

---

```suggestion
        hashable_items = []
        for item in self:
            hashable_items.extend((key, frozenset(value)) for key, value in item.items())
        
        return hash(frozenset(hashable_items))
```

Reply:
good catch!